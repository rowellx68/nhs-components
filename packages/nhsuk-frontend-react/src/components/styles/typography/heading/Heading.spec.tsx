import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, { H1 as HeadingStory } from './Heading.stories';

const Heading = composeStory(HeadingStory, meta);

it.each(['h1', 'h2', 'h3', 'h4', 'h5', 'h6'])(
  'should render the %s Heading component',
  (heading) => {
    const { container } = render(<Heading as={heading} />);

    expect(container.querySelector(heading)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  },
);

it.each(['xs', 's', 'm', 'l', 'xl'])(
  'should render the %s Heading component',
  (size) => {
    const { container } = render(<Heading size={size} />);

    expect(
      container.querySelector(`.nhsuk-heading-${size}`),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  },
);
