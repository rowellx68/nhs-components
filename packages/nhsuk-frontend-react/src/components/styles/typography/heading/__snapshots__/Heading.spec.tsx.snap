// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the h1 Heading component 1`] = `
<div>
  <h1
    class="nhsuk-heading-xl"
  >
    nhsuk-heading-
    xl
  </h1>
</div>
`;

exports[`should render the h2 Heading component 1`] = `
<div>
  <h2
    class="nhsuk-heading-xl"
  >
    nhsuk-heading-
    xl
  </h2>
</div>
`;

exports[`should render the h3 Heading component 1`] = `
<div>
  <h3
    class="nhsuk-heading-xl"
  >
    nhsuk-heading-
    xl
  </h3>
</div>
`;

exports[`should render the h4 Heading component 1`] = `
<div>
  <h4
    class="nhsuk-heading-xl"
  >
    nhsuk-heading-
    xl
  </h4>
</div>
`;

exports[`should render the h5 Heading component 1`] = `
<div>
  <h5
    class="nhsuk-heading-xl"
  >
    nhsuk-heading-
    xl
  </h5>
</div>
`;

exports[`should render the h6 Heading component 1`] = `
<div>
  <h6
    class="nhsuk-heading-xl"
  >
    nhsuk-heading-
    xl
  </h6>
</div>
`;

exports[`should render the l Heading component 1`] = `
<div>
  <h1
    class="nhsuk-heading-l"
  >
    nhsuk-heading-
    l
  </h1>
</div>
`;

exports[`should render the m Heading component 1`] = `
<div>
  <h1
    class="nhsuk-heading-m"
  >
    nhsuk-heading-
    m
  </h1>
</div>
`;

exports[`should render the s Heading component 1`] = `
<div>
  <h1
    class="nhsuk-heading-s"
  >
    nhsuk-heading-
    s
  </h1>
</div>
`;

exports[`should render the xl Heading component 1`] = `
<div>
  <h1
    class="nhsuk-heading-xl"
  >
    nhsuk-heading-
    xl
  </h1>
</div>
`;

exports[`should render the xs Heading component 1`] = `
<div>
  <h1
    class="nhsuk-heading-xs"
  >
    nhsuk-heading-
    xs
  </h1>
</div>
`;
